htdocs/mrp/class/mo.class.php

```diff
@@ -1648,4 +1667,92 @@ public function delete(User $user, $notrigger = false)
 		return $this->deleteCommon($user, $notrigger);
 		//return $this->deleteCommon($user, $notrigger, 1);
 	}
+
+	/**
+	 *  Return a link to the object card (with optionaly the picto)
+	 *
+	 *  @param  int     $withpicto                  Include picto in link (0=No picto, 1=Include picto into link, 2=Only picto)
+	 *  @param  string  $option                     On what the link point to ('nolink', '', 'production', ...)
+	 *  @param  int     $notooltip                  1=Disable tooltip
+	 *  @param  string  $morecss                    Add more css on link
+	 *  @param  int     $save_lastsearch_value      -1=Auto, 0=No save of lastsearch_values when clicking, 1=Save lastsearch_values whenclicking
+	 *  @return	string                              String with URL
+	 */
+	public function getNomUrl($withpicto = 0, $option = '', $notooltip = 0, $morecss = '', $save_lastsearch_value = -1)
```

Comment:
Here the method getNomUrl is inside the MoLine. So it MUST return the ref of a line.
But a MO line has no ref, except its id.
You don't need to add such a method here. 
You mus tuse the existing method getNomUrl that exists on MO object to show a link to a MO.

The fk_parent_line must have a type that describe it is a foreign key to a MO object.